docs/use_cases/knowledge_graph_embedding.md

```diff
@@ -0,0 +1,228 @@
+
+## What are Knowledge Graphs?
+
+Now, let's zoom in on KGs (Knowledge Graphs). We use KGs to describe how different entities, like people, places, or more generally "things", relate to each other. For example a KG can show us how a famous writer is linked to their books or how a book is connected to its received awards:
+
+![Knowledge Graph example](../assets/use_cases/knowledge_graph_embedding/small_kg.png)
+
+In certain areas where understanding these specific connections is crucial - like recommendation systems, search engines, information retrieval, etc. - KGs step in as specialized tools. They help computers grasp the detailed relationships between things.
+
+## What is Knowledge Graph Embedding (KGE)?
+
+KGE algorithms take this tangled complex web of connections and turn it into something AI systems can understand better: vectors. This might raise a question - if we already have knowledge of the connections between nodes and their relations, then why do we need to take the effort to learn embeddings?
+
+The challenge with KGs is that they are usually incomplete. This means that there might be some edges that should ideally be present but are missing. These missing links could be the result of inaccuracies in the data collection process, or it could simply be a reflection of the imperfect nature of our data source. According to [this article](https://towardsdatascience.com/neural-graph-databases-cc35c9e1d04f), in large open-source knowledge bases we can observe a significant amount of incompleteness: 
+
+“… in Freebase, 93.8% of people have no place of birth and [78.5% have no nationality](https://aclanthology.org/P09-1113.pdf), about 68% of people [do not have any profession](https://dl.acm.org/doi/abs/10.1145/2566486.2568032), while in Wikidata, about [50% of artists have no date of birth](https://arxiv.org/abs/2207.00143), and only [0.4% of known buildings have information about height](https://dl.acm.org/doi/abs/10.1145/3485447.3511932).”
+
+These imperfections, whether minor or major, can pose significant difficulties if we solely rely on the graph for information. In such a scenario, KGE algorithms prove to be extremely beneficial. They allow us to retrieve the missing data by utilizing the learned semantic meaning of nodes and links.
+
+### How KGE algorithms work?
+
+In general, KGE algorithms work by defining a similarity function in the embedding space, and learn the model by minimizing a loss function that penalizes the discrepancy between this similarity function and some notion of similarity in the graph. KGE algorithms can differ in the choice of the similarity function and the definition of node similarity in the graph.
+
+The simplest approach is to consider nodes that are connected by an edge as similar. Then, the task of learning node embeddings can be defined as a classification task. Given the embeddings of two nodes and a relation, the task is to determine how likely it is that they are similar (connected).
```

Comment:
I don't get this.Isn't it the thing that connected nodes' embeddings should be similar to each other?

Reply:
It means exactly that. By using dot product as the similarity function, and cross entropy on the presence of an edge as the loss, that will force connected nodes to have similar embeddings (high dot product value) while not connected pairs to have low similarity (low dot product).

Reply:
I get it now :) I think you should extend this paragraph to include this, too.